apps/generator/lib/conditionalGeneration.js

```diff
@@ -0,0 +1,267 @@
+const fs = require('fs').promises;
+const path = require('path');
+const log = require('loglevel');
+const logMessage = require('./logMessages');
+const jmespath = require('jmespath');
+
+/**
+ * Determines whether the generation of a file or folder should be skipped
+ * based on conditions defined in the template configuration.
+ *
+ * @param {string} relativeSourceFile - The relative path of the source file.
+ * @param {string} relativeSourceDirectory - The relative path of the source directory.
+ * @param {string} relativeTargetFile - The relative path of the target file to be generated.
+ * @param {Object} templateConfig - The template configuration containing conditional logic.
+ * @param {string} targetDir - The directory where the generated files are written.
+ * @param {string} matchedConditionPath - The matched path used to find applicable conditions.
+ * @param {Object} templateParams - Parameters passed to the template.
+ * @param {AsyncAPIDocument} asyncapiDocument - The AsyncAPI document used for evaluating conditions.
+ * @returns {Promise<boolean>} A promise that resolves to `true` if generation should be skipped, otherwise `false`.
+ */
+async function conditionalGeneration (
```

Comment:
wouldn't it be better if the main function used in generator.js is more intuitive in name, like `isGenerationConditionMet` or someting, and `true` would be that yes condition is met, so file/folder shoudl go through rendering, and if condition is not met, then return `false` and files are skipped

Reply:
basically files that meet condition, like for example

```
"conditionalOnFile.js": { 
       "parameter": "singleFile", 
       "validation": {
          "not": { "const": true } 
      }
    }
```

if condition is met, `singleFile` was passed with `false`, then `conditionalOnfile.js` shoudl be rendered to output, otherwise it should be skipped


btw `"const": true` example needs a fix, true must be as string. Remember that all stuff that goes through parameters in CLI is anyway string, even Boolean

Reply:
Updated and changed the method name @derberg